### PR TITLE
WIP - Update scilpy & tractometry

### PR DESCRIPTION
**Pull request by frheault**
Adaptation to the recent modification in scilpy

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -57,7 +57,7 @@ log.info "Input: $params.root"
 root = file(params.root)
 /* Watch out, files are ordered alphabetically in channel */
     in_data = Channel
-        .fromFilePairs("$root/**/{*tracking.trk,*anat.nii.gz}",
+        .fromFilePairs("$root/**/{*tracking.trk,*fa.nii.gz}",
```
Comment (alexjoa) [inline main.nf]: The name anat should be changed to fa in USAGE too.

File: main.nf [hunk 2]
```diff
@@ -56,29 +56,28 @@ log.info ""
...
+    .fromPath("$root/**/*tracking*.trk",
+                    maxDepth:1)
+    .map{[it.parent.name, it]}
+    .set{tractogram_for_recognition}
+
+Channel
+    .fromPath("$root/**/*fa.nii.gz",
+                    maxDepth:1)
+    .map{[it.parent.name, it]}
+    .into{anat_for_registration;anat_for_reference_c;anat_for_reference_b}
```
Comment (GuillaumeTh) [inline main.nf]: I don't like variable naming. I prefer something like `anat_for_transform_centroids`

**Review by GuillaumeTh — COMMENTED** [review]
Don't tested yet. A small comment about naming convention. Don't tested yet. @frheault Could you give me the path of the singularity used for this PR or at least scilpy commit number ?

File: main.nf [hunk 3]
```diff
@@ -56,29 +56,28 @@ log.info ""
...
+
+atlas_anat = Channel.fromPath("$params.atlas_anat")
```
Comment (GuillaumeTh) [inline main.nf:76]: Could be nice to test if all the mandatory arguments are differents than false. Something like in TractoFlow just to be more robust.

**Review by GuillaumeTh — CHANGES_REQUESTED** [review]
The pipeline crash. Maybe a too old version of the models ?

```
File "/usr/local/lib/python3.7/dist-packages/scilpy-0.2.dev0-py3.7-linux-x86_64.egg/scilpy/segment/voting_scheme.py", line 73, in _init_bundles_tag
      for key in self.config.keys():
  AttributeError: 'list' object has no attribute 'keys'
```

**Review by GuillaumeTh — CHANGES_REQUESTED** [review]
Worked well with the atlas you gave me. Please fix the comments and I will approve after.

**Comment by frheault** [comment]
@GuillaumeTh Done

File: USAGE [hunk 4]
```diff
@@ -11,7 +11,7 @@ DESCRIPTION
 
                                     [root]
                                     ├── S1
-                                    │   ├── *anat.nii.gz
+                                    │   ├── *fa.nii.gz
```
Comment (mdesco) [inline USAGE:14]: Does it really need to be an FA? 
For instance, in my test case, I have b0 and I believe it should work, no?

Can you add something in the help about that please?

Comment (mdesco) [reply USAGE:14]: Also,

1. Can you provide a link to your zenodo URL please
2. Can you add a centroids directory or download on the zenodo?

Comment (mdesco) [reply USAGE:14]: Finally, a lot of our flows use --root, others use --input. Can we harmonize this?

--input or --root?

Comment (mdesco) [reply USAGE:14]: --input

**Review by mdesco — APPROVED** [review]
Aside from my comestic suggestions. This works and was tested with and without the singularity tractoflow-scilpy-rc2

File: USAGE (hunk 4 above)
Comment (frheault) [reply USAGE:14]: It should work with anything, but I want to leave *fa.nii.gz as an input name

**Review by GuillaumeTh — APPROVED** [review]
Tested and worked well. Good job ! :beers:

**Pull request merged**